.circleci/config.yml

```
@@ -102,23 +102,33 @@ jobs:
               if [[ -f "$subdir"/test_requirements.txt ]]; then
                 pip install -r "$subdir"/test_requirements.txt
               fi
+              if [[ -f "$subdir"/requirements_dev.txt ]]; then
+                pip install -r "$subdir"/requirements_dev.txt
+              fi
               done
-            pip install cython wheel
-            pip install .
+            pip install cython wheel cgen
+            pip install .[all]
             dwave install --all --yes
 
       - run:
-          name: run individual package unittests
+          name: run individual package unit tests
           command: |
             . env/bin/activate
             for subdir in */; do
               if [[ -d "$subdir"/tests ]] && [[ "$subdir" != "env/" ]]; then
                 cd "$subdir"
                 pwd
                 if [[ -f "setup.py" ]]; then
+                  if [[ "$subdir" == "dwave-gate/" ]]; then
+                    python dwave/gate/simulator/operation_generation.py
+                  fi
                   python setup.py build_ext --inplace 2>/dev/null
                 fi
-                python -m unittest discover
+                if [[ -f "requirements_dev.txt" ]] && grep -q pytest "requirements_dev.txt"; then
+                  python -m pytest tests/
+                else
+                  python -m unittest discover
+                fi
```

Why not just:
```suggestion
                python -m pytest tests
```

Pytest handles Python unittests OOTB. We would just need to always install `pytest` (few lines above).

---

Ah, didn't realize that. Much simpler then!

---

@randomir Tests now fail with `AttributeError: 'TestMarginals' object has no attribute 'bqm'` in `dwave-samplers`. 😱

---

Different test discovery patterns. `pytest` does by class name but `unittest` does by inheritance from `unittest.TestCase`

---

To be fair, it's an odd pattern in dwave-samplers, not sure why I wrote it that way (at least I think it was me). Probably should get a refactor for clarity at some point https://github.com/dwavesystems/dwave-samplers/blob/a16f7b9c929632a1807347a486bff43950d45d64/tests/test_tree_sampler.py#L110-L281

---

I'll just make a PR to dwave-samplers today. Should be quick and supporting other testing frameworks like `pytest` is a nice feature if we can manage it.

---

Yeah, renaming `TestMarginals` mixin should do the trick.

---

With https://github.com/dwavesystems/dwave-samplers/pull/49 merged, I think all you need to do is update the commit of dwave-samplers to fix the tests.

---

@arcondello Seems to be a similar issue [here ](https://github.com/dwavesystems/dwave-system/blob/master/tests/test_embedding_chain_breaks.py#L64)with dwave-system.

---

@arcondello @randomir `test_individual_pkgs` is now passing (along with all other checks). Should be ready for another review/approval. Thanks for the help with getting this in!